src/main/java/sirius/biz/storage/layer3/VirtualFile.java

```diff
@@ -0,0 +1,794 @@
+                childProvider.enumerate(this, search);
+            }
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "childProvider.enumerate");
+        }
+    }
+
+    /**
+     * Tries to find a child with the given name.
+     * <p>
+     * Note that the given file can be non-existent.
+     *
+     * @param name the name of the child to find
+     * @return the child wrapped as optional or an empty optional if the given name cannot be resolved into a file.
+     */
+    public Optional<VirtualFile> findChild(String name) {
+        try {
+            if (childProvider == null) {
+                return Optional.empty();
+            }
+
+            return childProvider.findChild(this, name);
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "childProvider.findChild");
+        }
+    }
+
+    /**
+     * Tries to resolve the relative path within this directory.
+     *
+     * @param relativePath the path to resolve
+     * @return the relative path wrapped as optional or an  empty optional if the given relaive path cannot be
+     * resolved into a file.
+     */
+    public Optional<VirtualFile> tryResolve(String relativePath) {
+        if (relativePath != null && relativePath.startsWith("/")) {
+            relativePath = relativePath.substring(1);
+        }
+
+        if (Strings.isEmpty(relativePath)) {
+            return Optional.empty();
+        }
+
+        Tuple<String, String> nameAndRest = Strings.split(relativePath, "/");
+        Optional<VirtualFile> child = findChild(nameAndRest.getFirst());
+        if (Strings.isFilled(nameAndRest.getSecond())) {
+            return child.flatMap(subChild -> subChild.tryResolve(nameAndRest.getSecond()));
+        }
+
+        return child;
+    }
+
+    /**
+     * Resolves the relative path within this directory.
+     *
+     * @param relativePath the path to resolve
+     * @return the relative path wrapped as optional or an  empty optional if the given relaive path cannot be
+     * resolved into a file.
+     * @throws HandledException if the given path cannot be resolved
+     */
+    public VirtualFile resolve(String relativePath) {
+        return tryResolve(relativePath).orElseThrow(() -> Exceptions.createHandled()
+                                                                    .withNLSKey("VirtualFile.cannotResolveChild")
+                                                                    .set("path", path())
+                                                                    .set("child", relativePath)
+                                                                    .handle());
+    }
+
+    /**
+     * Determines if children can be created for this file.
+     *
+     * @return <tt>true</tt> if children can (probably) be created, <tt>false</tt> otherwise
+     */
+    public boolean canCreateChildren() {
+        try {
+            if (canCreateChildrenHandler == null) {
```

Comment:
I think inverting this if check would improve readability/flow and would be consistent to the other methods in this class